### ISSUE TITLE
Client-side experimental_onFunctionCall not submitting message request

### ISSUE THREAD

**Issue by woodbridge** [vercel/ai#476]
I'm trying to get a basic client side implementation of a function working. Version: 2.2.4 (very latest)

Server-side
```
const functions: CompletionCreateParams.Function[] = [
  {
    name: "get_available_objects",
    description:
      "get a list of the objects that are available to insert into the editor",
    parameters: {
      type: "object",
      properties: {},
    },
  },
];

const openai = new OpenAI({
  apiKey: ...
});

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  const { messages } = JSON.parse(req.body);

  const response = await openai.chat.completions.create({
    model: "gpt-4",
    stream: true,
    messages: messages,
    functions,
  });

  const stream = OpenAIStream(response);

  return streamToResponse(stream, res);
}
```

Client-side
```
const functionCallHandler: FunctionCallHandler = async (
  chatMessages,
  functionCall
) => {
  if (functionCall.name === "get_available_objects") {
    const functionResponse: ChatRequest = {
      messages: [
        ...chatMessages,
        {
          id: nanoid(),
          name: "get_available_objects",
          role: "function" as const,
          content: JSON.stringify(["cube", "sphere", "porsche"]),
        },
      ],
    };
    console.log("Returning...", functionResponse);
    return functionResponse;
  }
};
```

Chat component
```
export const Panel = ({ show }: { show: boolean }) => {
   ...
  const {
    input,
    handleSubmit,
    messages,
    setMessages,
    handleInputChange,
    error,
  } = useChat({
    experimental_onFunctionCall: functionCallHandler,
  });

  console.log("MESSAGES", messages);
```

When I do a test run, the model tries to use the function, and the client-side handler is called, generating a function response that includes the correct contents ("returning..."), however, subsequently that message isn't included in the message list, and there's request to the chat endpoint including it.

<img width="1179" alt="image" src="https://github.com/vercel/ai/assets/883009/e3a5c1fe-4d18-4c2d-9170-a7471aada930">

Am I understanding the api correctly / missing something obvious?

**Comment by MaxLeiter** [comment]
Thanks for the report, I'll look into this. In the meantime, you can use `ai@2.2.0`.

**Comment by woodbridge** [comment]
@MaxLeiter thanks for the speedy ack + fix

**Comment by MaxLeiter** [comment]
It will be out in 2.2.5 👍

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a concrete regression in existing client-side function-calling behavior, acknowledged by a maintainer and scheduled for a patch release.

### Evidence
- In ai@2.2.4, experimental_onFunctionCall executes and returns a ChatRequest containing the function result, but useChat does not add that result to messages or submit the follow-up chat request.
- The maintainer recommended downgrading to ai@2.2.0 and stated the fix would ship in 2.2.5, confirming a regression.
